engine/src/ast/function_expr.rs

```diff
@@ -164,8 +164,15 @@ impl<'i, 's> LexWith<'i, &'s Scheme> for FunctionCallExpr<'s> {
 
         let mut index = 0;
 
-        while let Some(',') = input.chars().next() {
-            input = skip_space(take(input, 1)?.1);
+        while let Some(c) = input.chars().next() {
+            if c == ')' {
+                break;
+            }
+            if !function_call.args.is_empty() {
```

Comment:
This seems too specific. If you haven't seen `)`, then you should always expect `,`, no?

Reply:
Maybe i misunderstood your comment, but in case there are no mandatory arguments and the first argument is an optional one then you should not expect a `,` in this case. Doing so would lead to parse `func(,arg)` as valid.

Reply:
Then, instead, you should check `index` to make it more obvious that you don't expect a `,` before the first argument (when `index == 0`). And move skip_space inside the condition, since otherwise it's not necessary.

Reply:
`index  == 0` does only mean `first optional argument`, there could have been zero or more mandatory arguments. We only expect a `,` if the argument is not the first.

Reply:
Oh, okay. This seems easy to miss, can you leave a comment?

Reply:
Ok, i'll add one ASAP.

Reply:
Done